### PR TITLE
daemon: Fix session affinity map creation

### LINKED ISSUE #12118: CI: controller bpf-map-sync-cilium_lb_affinity_match is failing
Test **Endpoint can still connect while Cilium is not running** failed on `test-gke`:

```
11:54:38  18:54:37 STEP: Cilium is not ready yet: controllers are failing: cilium-agent 'cilium-d884n': controller bpf-map-sync-cilium_lb_affinity_match is failing: Exitcode: 0 
11:54:38  Stdout:
11:54:38   	 KVStore:                Ok   Disabled
11:54:38  	 Kubernetes:             Ok   1.14+ (v1.14.10-gke.36) [linux/amd64]
11:54:38  	 Kubernetes APIs:        ["CustomResourceDefinition", "cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Endpoint", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "networking.k8s.io/v1::NetworkPolicy"]
11:54:38  	 KubeProxyReplacement:   Probe   []   [SessionAffinity]
11:54:38  	 Cilium:                 Ok      OK
11:54:38  	 NodeMonitor:            Listening for events on 4 CPUs with 64x4096 of shared memory
11:54:38  	 Cilium health daemon:   Ok   
11:54:38  	 IPAM:                   IPv4: 2/255 allocated from 10.4.47.0/24, 
11:54:38  	 Masquerading:           IPTables
11:54:38  	 Controller Status:      17/18 healthy
11:54:38  	   Name                                    Last success   Last error   Count   Message
11:54:38  	   bpf-map-sync-cilium_lb_affinity_match   never          15s ago      23      4 map sync errors   
11:54:38  	   cilium-health-ep                        26s ago        never        0       no error            
11:54:38  	   dns-garbage-collector-job               27s ago        never        0       no error            
11:54:38  	   endpoint-1949-regeneration-recovery     never          never        0       no error            
11:54:38  	   endpoint-827-regeneration-recovery      never          never        0       no error            
11:54:38  	   k8s-heartbeat                           31s ago        never        0       no error            
11:54:38  	   mark-k8s-node-as-available              4m27s ago      never        0       no error            
11:54:38  	   metricsmap-bpf-prom-sync                6s ago         never        0       no error            
11:54:38  	   resolve-identity-1949                   4m26s ago      never        0       no error            
11:54:38  	   resolve-identity-827                    4m27s ago      never        0       no error            
11:54:38  	   sync-endpoints-and-host-ips             27s ago        never        0       no error            
11:54:38  	   sync-lb-maps-with-k8s-services          4m27s ago      never        0       no error            
11:54:38  	   sync-policymap-1949                     25s ago        never        0       no error            
11:54:38  	   sync-policymap-827                      26s ago        never        0       no error            
11:54:38  	   sync-to-k8s-ciliumendpoint (1949)       6s ago         never        0       no error            
11:54:38  	   sync-to-k8s-ciliumendpoint (827)        7s ago         never        0       no error            
11:54:38  	   template-dir-watcher                    never          never        0       no error            
11:54:38  	   update-k8s-node-annotations             4m28s ago      never        0       no error            
11:54:38  	 Proxy Status:     OK, ip 10.4.47.246, 0 redirects active on ports 10000-20000
11:54:38  	 Hubble:           Ok              Current/Max Flows: 336/4096 (8.20%), Flows/s: 1.27   Metrics: Disabled
11:54:38  	 Cluster health:   2/2 reachable   (2020-06-16T18:54:29Z)
```

### PR DESCRIPTION
**Pull request by brb**
This PR fixes the BPF session affinity -related maps creation, improves logging of auto detection and moves the replacement helpers into `cmd/kube_proxy_replacement.go`.

Reviewable per commit.

Fixes: bcdcf9b ("daemon: Move kubeProxyReplacement init after connect to k8s")
Reported-by: Jarno Rajahalme <jarno@covalent.io>

Fix #12118.

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Comment by brb** [comment]
retest-4.19

File: daemon/cmd/daemon_main.go [hunk 1]
```diff
@@ -1590,9 +1591,10 @@ func initKubeProxyReplacementOptions() (strict bool) {
 	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementProbe ||
 		option.Config.KubeProxyReplacement == option.KubeProxyReplacementStrict {
 
-		log.Infof("Auto-enabling %q, %q, %q, %q features", option.EnableNodePort,
-			option.EnableExternalIPs, option.EnableHostReachableServices,
-			option.EnableHostPort)
+		log.Infof("Trying to auto-enabling %q, %q, %q, %q, %q features",
```
Comment (joestringer) [inline daemon/cmd/daemon_main.go]:
```suggestion
		log.Infof("Trying to auto-enable %q, %q, %q, %q, %q features",
```

**Review by joestringer — APPROVED** [review]
Minor nit. Is the CI failure a flake or potentially caused by this PR?

**Comment by joestringer** [comment]
K8sDatapathConfig Etcd Check connectivity failure:

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Kernel/2162/testReport/junit/Suite-k8s-1/17/K8sDatapathConfig_Etcd_Check_connectivity/

**Comment by borkmann** [comment]
> K8sDatapathConfig Etcd Check connectivity failure:
> 
> https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Kernel/2162/testReport/junit/Suite-k8s-1/17/K8sDatapathConfig_Etcd_Check_connectivity/

Known CI flake on 4.19; I think not tracked yet though.

**Comment by joestringer** [comment]
retest-4.19

**Comment by brb** [comment]
4.19 hit the same Etcd flake.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
Hit `Suite-k8s-1.11.K8sPolicyTest Multi-node policy test validates fromEntities policies with remote-node identity enabled Validates fromEntities remote-node policy` (https://jenkins.cilium.io/job/Cilium-PR-K8s-oldest-net-next/869/testReport/junit/Suite-k8s-1/11/K8sPolicyTest_Multi_node_policy_test_validates_fromEntities_policies_with_remote_node_identity_enabled_Validates_fromEntities_remote_node_policy/)

**Comment by brb** [comment]
retest-net-next

**Comment by brb** [comment]
test-me-please

**Pull request merged**